src/main/java/uk/ac/ebi/atlas/experimentpage/JsonMarkerGenesController.java

```diff
@@ -27,12 +28,12 @@ public JsonMarkerGenesController(MarkerGenesDao markerGenesDao,
     }
 
     @Deprecated
-    @GetMapping(value = "/json/experiments/{experimentAccession}/marker-genes/{k}",
+    @GetMapping(value = "/json/experiments/{experimentAccession}/marker-genes/clusters",
                 produces = MediaType.APPLICATION_JSON_UTF8_VALUE)
     public String getMarkerGenes(@PathVariable String experimentAccession,
-                                 @PathVariable int k) {
+                                 @RequestParam String k) {
         return GSON.toJson(
-                highchartsHeatmapAdapter.getMarkerGeneHeatmapData(
+                highchartsHeatmapAdapter.getCellTypeMarkerGeneHeatmapData(
```

Comment:
If this is a generic controller for both the old marker gene heatmap and the new cell type marker gene heatmap, I think this controller should have a more generic name, such as `getCellTypeAndClusterMarkerGeneHeatmapData`.

Reply:
- Made this HearMapAdapter class method signature more generic like` getMarkerGeneHeatmapData`.

- JsonMarkerGeneController-  methods I have made a bit more specific as we are using the specific methods for cluster and cell type marker genes